### ISSUE TITLE
[Bug]: kokkos test for E4S 21.11 requires cmake

### ISSUE THREAD

**Issue by shahzebsiddiqui** [buildtesters/buildtest-nersc#125]
### CDASH Build

https://my.cdash.org/test/61574963

### Link to buildspec file

https://github.com/buildtesters/buildtest-nersc/blob/devel/buildspecs/e4s/spack_test/perlmutter/kokkos.yml

### Please describe the issue?

The E4S 21.11 stack on Perlmutter requires rebuild similar to #124. Once stack is rebuilt test may pass but we may want to make sure we test it locally. I suspect test needs cmake so we may want to see whether `spack load cmake` would do the trick.  

### Relevant log output

```shell
==> Error: ProcessError: Command exited with status 1:
    'cmake' '/global/common/software/spackecp/perlmutter/e4s-21.11/software/cray-sles15-zen3/gcc-11.2.0/kokkos-3.4.01-7a2enxire5aqgcqlpl5b5bu6qglghp4x/.spack/test/scripts/spack_test/out' '-DEXECUTABLE_OUTPUT_PATH=/global/common/software/spackecp/perlmutter/e4s-21.11/software/cray-sles15-zen3/gcc-11.2.0/kokkos-3.4.01-7a2enxire5aqgcqlpl5b5bu6qglghp4x/.spack/test/scripts/spack_test/out'
4 errors found in test log:
     7     -- Detecting CXX compiler ABI info
     8     -- Detecting CXX compiler ABI info - done
     9     -- Detecting CXX compile features
     10    -- Detecting CXX compile features - done
     11    -- Enabled Kokkos devices: OPENMP;SERIAL
     12    -- Configuring done
  >> 13    CMake Error at CMakeLists.txt:18 (target_link_libraries):
     14      Error evaluating generator expression:
     15    
     16        $<LINK_LANGUAGE:CXX>
     17    
     18      Expression did not evaluate to a known generator expression
     19    
     20    
  >> 21    CMake Error at CMakeLists.txt:18 (target_link_libraries):
     22      Error evaluating generator expression:
     23    
     24        $<LINK_LANGUAGE:CXX>
     25    
     26      Expression did not evaluate to a known generator expression
     27    
     28    
  >> 29    CMake Error at CMakeLists.txt:18 (target_link_libraries):
     30      Error evaluating generator expression:
     31    
     32        $<LINK_LANGUAGE:CXX>
     33    
     34      Expression did not evaluate to a known generator expression
     35    
     36    
  >> 37    CMake Error at CMakeLists.txt:18 (target_link_libraries):
     38      Error evaluating generator expression:
     39    
     40        $<LINK_LANGUAGE:CXX>
     41    
     42      Expression did not evaluate to a known generator expression
     43    
See test log for details:
  /global/homes/e/e4s/.spack/test/5q7zu7bne6rfgmf2mpvnzelu3n5ciae2/kokkos-3.4.01-7a2enxi-test-out.txt
==> Error: IndexError: list index out of range
/global/common/software/spackecp/perlmutter/e4s-21.11/spack/var/spack/repos/builtin/packages/cray-mpich/package.py:104, in headers:
        102    def headers(self):
        103        hdrs = find_headers('mpi', self.prefix.include, recursive=True)
  >>    104        hdrs.directories = os.path.dirname(hdrs[0])
        105        return hdrs
See test log for details:
  /global/homes/e/e4s/.spack/test/5q7zu7bne6rfgmf2mpvnzelu3n5ciae2/kokkos-3.4.01-d75nvmj-test-out.txt
==> Error: 2 test(s) in the suite failed.
```

**Comment by etpalmer63** [comment]
It seems like Kokkos is failing to build its own smoke test. At first glance this looks like an issue with Kokkos own Spack smoke test.  I will test it locally.

**Comment by etpalmer63** [comment]
Based on @wspear 's suggestion, try spack load kokkos before running the tests because environment variables may not be set up correctly without doing this first.

**Comment by etpalmer63** [comment]
Both CPU and GPU tests with Spack v.019  run correctly on my local machine.

**Comment by shahzebsiddiqui** [comment]
The `spack test run kokkos` should run all test for all specs.

You can constraint the specs so `spack test run kokkos +cuda` to run the Cuda enabled tests. This all depends on if we have package installed

**Comment by shahzebsiddiqui** [comment]
Oh I just realized we made change to e4s module file where `e4s` spack env is not loaded so that means we need to do

`spack env activate -V e4s` followed by `spack load`.  This change was introduced recently to allow user to install software via spack without being tied to production stack. This will impact all e4s testsuite and spack test. @wspear you will want to update all the e4s testsuite tests as well. 

There is a difference. We have a spack view in 22.05 so it's best to do `spack env activate -V` so it doesn't enable view.

In 21.11 we have no view so it won't matter. The default behavior of `spack env activate ` is to enable view if it's present
